tests/core/test_db.py

```diff
@@ -0,0 +1,64 @@
+import pytest
+import pytest_asyncio
+
+from lnbits.db import QueryValues
+
+
+@pytest_asyncio.fixture(autouse=True)
+async def json_test_table(db):
+    try:
+        await db.execute("CREATE TABLE test (data JSON)")
+        yield db
+    finally:
+        await db.execute("DROP TABLE test")
+
+
+@pytest.mark.asyncio
+async def test_sql_json(db):
+    obj = {"a": 3, "b": "bar", "c": {"nested": "d"}}
+    another_obj = {"a": 2, "b": "baz"}
+
+    await db.execute("INSERT INTO test VALUES(?)", (obj,))
+    await db.execute("INSERT INTO test VALUES(?)", (another_obj,))
+
+    values = QueryValues()
+    rows = await db.fetchall(
+        f"""
+        SELECT * FROM test WHERE {values.json_path('data', 'a', type_=int)} >= {values(obj["a"])}
+        """,
+        values,
+    )
+    assert len(rows) == 1
+    assert rows[0].data == obj
+
+    values = QueryValues()
+    rows = await db.fetchall(
+        f"SELECT * FROM test WHERE {values.json_path('data', 'c', 'nested')} = {values(obj['c']['nested'])}",
+        values,
+    )
+    assert len(rows) == 1
+    assert rows[0].data == obj
+
+    values = QueryValues()
+    rows = await db.fetchall(
+        f"SELECT * FROM test WHERE {values.json_path('data', 'b')} != {values(obj['b'])}",
+        values,
+    )
+    assert len(rows) == 1
+    assert rows[0].data == another_obj
+
+    update = {"b": "baz"}
+    obj |= update
+    values = QueryValues()
+    await db.execute(
+        f"UPDATE test SET {values.json_partial_update('data', update)}", values
+    )
+    row = await db.fetchone("SELECT * FROM test")
+    assert row.data == obj
```

Comment:
-now there are two rows in the table,  this test passes by chance. It might as well return the `another_obj` row

Reply:
Isn't that test somewhat useless now? it essentially tests that the data is being parsed correctly which the above do aswell

Reply:
> it essentially tests that the data is being parsed correctly which the above do aswell
- it tests that the data is actually updated
- if `row = await db.fetchone("SELECT * FROM test")` would return the row containing `another_obj` then the test would fail
- alternatively, if you switch the order of the inserts from
```py
    await db.execute("INSERT INTO test VALUES(?)", (obj,))
    await db.execute("INSERT INTO test VALUES(?)", (another_obj,))
```
  - to
```py
    await db.execute("INSERT INTO test VALUES(?)", (another_obj,))
    await db.execute("INSERT INTO test VALUES(?)", (obj,))
```
 - then the test will fail
 - this can lead to unpredictable test success/failure in the CI builds (sometimes tests just fail apparently for no reason)